src/coreclr/jit/gentree.cpp

```diff
@@ -21973,6 +21972,40 @@ GenTree* Compiler::gtNewSimdBinOpNode(
                     break;
                 }
 
+                case TYP_LONG:
+                case TYP_ULONG:
+                {
+                    if (simdSize == 8)
+                    {
+                        // Vector64<long> vec = Vector64.CreateScalar(op1.ToScalar() * op2.ToScalar())
+                        op1            = gtNewSimdToScalarNode(TYP_LONG, op1, simdBaseJitType, 8);
+                        op2            = gtNewSimdToScalarNode(TYP_LONG, op2, simdBaseJitType, 8);
+                        GenTreeOp* mul = gtNewOperNode(GT_MUL, TYP_LONG, op1, op2);
+                        return gtNewSimdCreateScalarNode(TYP_SIMD8, mul, simdBaseJitType, 8);
+                    }
+
+                    // Make op1 and op2 multi-use:
+                    GenTree* op1Dup = fgMakeMultiUse(&op1);
+                    GenTree* op2Dup = fgMakeMultiUse(&op2);
+
+                    // long left0  = op1.GetElement(0)
+                    // long right0 = op2.GetElement(0)
+                    GenTree* left0  = gtNewSimdToScalarNode(TYP_LONG, op1, simdBaseJitType, 16);
+                    GenTree* right0 = gtNewSimdToScalarNode(TYP_LONG, op2, simdBaseJitType, 16);
+
+                    // long left1  = op1.GetElement(1)
+                    // long right1 = op2.GetElement(1)
+                    GenTree* left1  = gtNewSimdGetElementNode(TYP_LONG, op1Dup, gtNewIconNode(1), simdBaseJitType, 16);
+                    GenTree* right1 = gtNewSimdGetElementNode(TYP_LONG, op2Dup, gtNewIconNode(1), simdBaseJitType, 16);
+
+                    // Vector128<long> vec = Vector128.Create(left0 * right0, left1 * right1)
+                    op1          = gtNewOperNode(GT_MUL, TYP_LONG, left0, right0);
+                    op2          = gtNewOperNode(GT_MUL, TYP_LONG, left1, right1);
+                    GenTree* vec = gtNewSimdCreateScalarUnsafeNode(type, op1, simdBaseJitType, 16);
+                    return gtNewSimdHWIntrinsicNode(type, vec, gtNewIconNode(1), op2, NI_AdvSimd_Insert,
+                                                    simdBaseJitType, 16);
```

Comment:
nit: Use `gtNewSimdWithElementNode(type, vec, gtNewIconNode(1), op2, simdBaseJitType, simdSize)` which ensures all the optimal handling takes place.

Reply:
Thanks! Applied